components/raftstore/src/store/peer_storage.rs

```diff
@@ -599,10 +600,35 @@ fn init_apply_state<EK: KvEngine, ER: RaftEngine>(
         let apply_state = RaftApplyState::default();
         return Ok((None, apply_state));
     }
-    let tablet = engines.tablets.open_tablet(region.get_id(), suffix);
 
-    let apply_state = match tablet.get_msg_cf(CF_RAFT, &keys::apply_state_key(region.get_id()))? {
-        Some(s) => s,
+    let tablet = engines.tablets.open_tablet(region.get_id(), suffix);
+    let apply_state = match tablet
+        .get_msg_cf::<RaftApplyState>(CF_RAFT, &keys::apply_state_key(region.get_id()))?
+    {
+        Some(s) => {
+            if suffix == s.get_applied_index() {
```

Comment:
Maybe we need to always check two states and choose the maximun truncated index.